zipline/assets/roll_finder.py

```diff
@@ -212,6 +212,8 @@ def _active_contract(self, oc, front, back, dt):
             tc.minute_to_session_label(gap_end),
         )
         for session in sessions:
+            if session < back_contract.start_date:
+                continue
```

Comment:
What would you think of incorporating this logic when calculating `sessions` above? Something like making `gap_start` the max of the beginning of the grace period and the back contract start date?